awx/main/migrations/0064_notification_timeout.py

```diff
@@ -0,0 +1,20 @@
+# -*- coding: utf-8 -*-
```

Comment:
On closer inspection, I don't think we actually _need_ this field (or migration) for the timeout to work.  We don't generally add backend-specific fields like this to the `main_notification` table unless all of the notifications use them (similarly, `main_notification` doesn't have a username/password/host/port column).

With this diff:

```diff
diff --git a/awx/api/serializers.py b/awx/api/serializers.py
index 7dead78ea..ca454a0b1 100644
--- a/awx/api/serializers.py
+++ b/awx/api/serializers.py
@@ -4649,7 +4649,7 @@ class NotificationSerializer(BaseSerializer):
     class Meta:
         model = Notification
         fields = ('*', '-name', '-description', 'notification_template', 'error', 'status', 'notifications_sent',
-                  'notification_type', 'recipients', 'subject', 'timeout')
+                  'notification_type', 'recipients', 'subject')

     def get_related(self, obj):
         res = super(NotificationSerializer, self).get_related(obj)
diff --git a/awx/main/migrations/0064_notification_timeout.py b/awx/main/migrations/0064_notification_timeout.py
deleted file mode 100644
index 19f90eb04..000000000
--- a/awx/main/migrations/0064_notification_timeout.py
+++ /dev/null
@@ -1,20 +0,0 @@
-# -*- coding: utf-8 -*-
-# Generated by Django 1.11.16 on 2019-03-08 20:55
-from __future__ import unicode_literals
-
-from django.db import migrations, models
-
-
-class Migration(migrations.Migration):
-
-    dependencies = [
-        ('main', '0063_v350_org_host_limits'),
-    ]
-
-    operations = [
-        migrations.AddField(
-            model_name='notification',
-            name='timeout',
-            field=models.IntegerField(blank=True, default=0, help_text='The amount of time (in seconds) to run before the notification is canceled.'),
-        ),
-    ]
diff --git a/awx/main/migrations/0065_auto_20190312_1658.py b/awx/main/migrations/0065_auto_20190312_1658.py
deleted file mode 100644
index dd2edeae3..000000000
--- a/awx/main/migrations/0065_auto_20190312_1658.py
+++ /dev/null
@@ -1,20 +0,0 @@
-# -*- coding: utf-8 -*-
-# Generated by Django 1.11.16 on 2019-03-12 16:58
-from __future__ import unicode_literals
-
-from django.db import migrations, models
-
-
-class Migration(migrations.Migration):
-
-    dependencies = [
-        ('main', '0064_notification_timeout'),
-    ]
-
-    operations = [
-        migrations.AlterField(
-            model_name='notification',
-            name='timeout',
-            field=models.IntegerField(blank=True, default=30, help_text='The amount of time (in seconds) to run before the notification is canceled.'),
-        ),
-    ]
diff --git a/awx/main/models/notifications.py b/awx/main/models/notifications.py
index 6bd3e4c4e..39cea4917 100644
--- a/awx/main/models/notifications.py
+++ b/awx/main/models/notifications.py
@@ -185,11 +185,6 @@ class Notification(CreatedModifiedModel):
         default='',
         editable=False,
     )
-    timeout = models.IntegerField(
-        blank=True,
-        default=30,
-        help_text=_("The amount of time (in seconds) to run before the notification is canceled."),
-    )
     body = JSONField(blank=True)

     def get_absolute_url(self, request=None):
```

The timeout functionality seems to still work fine for me.

![image](https://user-images.githubusercontent.com/214912/54297572-883e4b80-458d-11e9-95eb-a8aa9724c4b3.png)

Reply:
I got the same results, so this migration file has been removed.